cmd/gateway/main.go

```diff
@@ -124,7 +125,18 @@ func main() {
 	}
 	revproxy.RegisterHandlers(e, gwMiddlewares...)
 	// Initialize login server
-	loginServer, err := login.NewLoginServer(login.WithConfig(gwConfig.Login), login.WithSessionStore(sessionStore), login.WithTokenStore(tokenStore))
+	loginOptions := []login.LoginServerOption{login.WithConfig(gwConfig.Login), login.WithSessionStore(sessionStore), login.WithTokenStore(tokenStore)}
+
+	metricsClient, err := metrics.NewPosthogClient(gwConfig.Posthog)
+	if err != nil {
+		slog.Error("posthog client initializtion failed", "error", err)
+		os.Exit(1)
+	}
+	if metricsClient != nil {
+		defer metricsClient.Close()
+	}
+	loginOptions = append(loginOptions, login.WithMetricsClient(metricsClient))
+	loginServer, err := login.NewLoginServer(loginOptions...)
```

Comment:
Suggestion (nitpick):

```suggestion
	loginServer, err := login.NewLoginServer(login.WithConfig(gwConfig.Login), login.WithSessionStore(sessionStore), login.WithTokenStore(tokenStore), login.WithMetricsClient(metricsClient))
```